TidepoolKit/Model/Datum/TCalculatorDatum.swift

```diff
@@ -0,0 +1,100 @@
+//
+//  TCalculatorDatum.swift
+//  TidepoolKit
+//
+//  Created by Darin Krauss on 11/1/19.
+//  Copyright © 2019 Tidepool Project. All rights reserved.
+//
+
+import Foundation
+
+public class TCalculatorDatum: TDatum, Decodable {
+    public typealias BloodGlucoseTarget = TBloodGlucose.Target
+    public typealias Units = TBloodGlucose.Units
+
+    public var insulinOnBoard: Double?
+    public var bloodGlucoseInput: Double?
+    public var insulinSensitivity: Double?
+    public var carbohydrateInput: Double?
+    public var insulinCarbohydrateRatio: Double?
+    public var bloodGlucoseTarget: BloodGlucoseTarget?
+    public var recommended: Recommended?
+    public var bolus: TBolusDatum?      // NOTE: API write only
+    public var bolusId: String?         // NOTE: API read only
```

Comment:
Just want to note that I don't understand the process related to the bolus ID. Probably clearer if I understand the backend better. Confusion come from not seeing bolusID in bolus classes, so I assume this is server generated? Then on a read, the bolus ID can be used to pull the specific bolus details? Is there no need to cache the ID locally if the bolus details are updated?

Reply:
Yea, it is odd. The Tidepool data model documentation describes how this all works. 

Honestly, this is better than how the actual API works. The API (aka Tidepool data model) uses the *same* field name ('bolus') for *both* the bolus datum (a `struct` on input) and the bolus id (a `string` on output). This was included in the very first data model before my time. This is backend tech debt that has not yet been addressed.